### ISSUE TITLE
Best practice for using VERSION file when inside a git repo?

### ISSUE THREAD

**Issue by ronawho** [jemalloc/jemalloc#811]
When inside another git repo, what's the best way to force configure to use the version listed in the VERSION file? I was thinking of doing something like `--with-version=$(cat ${JEMALLOC_DIR}/VERSION)`, but that feels a little wonky, so I wanted to check if I'm missing some obvious solution.

I'm also curious if there's any reason the official releases don't just always use the VERSION file.

On a related note, the `--with-version` text in [INSTALL](https://github.com/jemalloc/jemalloc/blob/dev/INSTALL#L40) is little hard to parse.

**Comment by ronawho** [comment]
Sorry if you're seeing this question twice, I tried tacking onto https://github.com/jemalloc/jemalloc/issues/305, but I can never remember if github pings you for comments on closed issues.

**Comment by jasone** [comment]
Whoops, I looked into this when you first asked the question, but didn't post a response.  ```--with-version``` was created as a workaround to failed attempts to use ```git describe```, but apparently with the expectation that no ```VERSION``` file would be present.  Your proposal (```--with-version=$(cat ${JEMALLOC_DIR}/VERSION)```) is a reasonable approach.  Alternatively, we could modify ```with-version``` to accept ```--with-version=VERSION```, with the same effect.

**Comment by ronawho** [comment]
👍 `--with-version=VERSION` works for me


I'm happy to use `--with-version`, but it was certainly confusing until I figured out what was going on. Any reason to not just always use the VERSION file for official releases to prevent others from running into this issue altogether?
